When properly instructed, configure will figure out that the glob
implementation FreeBSD does not suffice. Unfortunately goa adds
the libc includes before the ones generated by configure. Use a
somewhat hackish approach to point to the proper header locally
and fix the structural issue another day.
--- src/local_glob.h
+++ src/local_glob.h
@@ -0,0 +1 @@
+#include_next "openbsd-compat/include/glob.h"
--- src/readconf.c
+++ src/readconf.c
@@ -27,7 +27,7 @@
 
 #include <ctype.h>
 #include <errno.h>
-#include <glob.h>
+#include "local_glob.h"
 #include <ifaddrs.h>
 #include <limits.h>
 #include <netdb.h>
--- src/scp.c
+++ src/scp.c
@@ -86,7 +86,7 @@
 #ifdef HAVE_FNMATCH_H
 #include <fnmatch.h>
 #endif
-#include <glob.h>
+#include "local_glob.h"
 #include <libgen.h>
 #include <locale.h>
 #include <poll.h>
--- src/servconf.c
+++ src/servconf.c
@@ -27,7 +27,7 @@
 #endif
 
 #include <ctype.h>
-#include <glob.h>
+#include "local_glob.h"
 #include <netdb.h>
 #include <pwd.h>
 #include <stdio.h>
--- src/sftp-glob.c
+++ src/sftp-glob.c
@@ -21,7 +21,7 @@
 #include <sys/stat.h>
 
 #include <dirent.h>
-#include <glob.h>
+#include "local_glob.h"
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
--- src/sftp-usergroup.c
+++ src/sftp-usergroup.c
@@ -21,7 +21,7 @@
 #include <sys/types.h>
 #include <sys/tree.h>
 
-#include <glob.h>
+#include "local_glob.h"
 #include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
--- src/sftp.c
+++ src/sftp.c
@@ -26,7 +26,7 @@
 
 #include <ctype.h>
 #include <errno.h>
-#include <glob.h>
+#include "local_glob.h"
 #include <paths.h>
 #include <libgen.h>
 #ifdef HAVE_LOCALE_H
